core/build.sh

```diff
@@ -64,11 +66,28 @@ Libraries will be built in 'build/<target>/<configuration>/src'.
   "uninstall       uninstall libraries from current system" \
   "${archtargets[@]}" \
   "--debug,-d                      configuration is 'debug', not 'release'" \
+  "--no-tests      do not configure tests (used by other projects)" \
   "--target-path=opt_target_path   override for build/ target path" \
   "--test=opt_tests,-t             test[s] to run (space separated)"
 
 builder_parse "$@"
 
+#
+# meson forces us to configure tests, including building compilers, even
+# if we don't plan to run them, for example when doing a dependency build
+# in CI
+#
+MESON_OPTION_keyman_core_tests=
+BUILD_BAT_keyman_core_tests=
+
+if builder_is_dep_build || builder_has_option --no-tests; then
+  MESON_OPTION_keyman_core_tests="-Dkeyman_core_tests=false"
+  BUILD_BAT_keyman_core_tests=--no-tests
+  builder_remove_dep /common/tools/hextobin
+  builder_remove_dep /common/web/keyman-version
+  builder_remove_dep /developer/src/kmc
+fi
```

Comment:
Noticed this little sneaky addition.  I was about to ask why it wasn't being done the same way as before - with a defined array added to `builder_describe` - but I suppose that wouldn't really work with the `--no-tests` option above, as `builder_has_option` requires that `builder_parse` has executed.

Reply:
Yeah, this is perhaps not perfect but there is a chicken-and-egg issue otherwise, so I opted to solve it this way.